classes/03class/docker/README.md

```diff
@@ -0,0 +1,261 @@
+# Docker
+
+This class is an introduction to Containers and Docker.
+The main goal is to explain the purpose of containers, the benefits and how to use it.
+We'll also take a look into how the AWs ECS Service works and it's purpose.
+
+***Contents***
+
+- [Docker](#docker)
+- [Containers](#containers)
+  - [What is a container?](#what-is-a-container)
+  - [Difference between Virtual Machine and Container](#difference-between-virtual-machine-and-container)
+  - [Benefits of COntainers](#benefits-of-containers)
+    - [Container](#container)
+    - [Virtual Machine](#virtual-machine)
+  - [What is Docker?](#what-is-docker)
+- [Respositories](#repositories)
+- [Dockerfile](#dockerfile)
+  - [Main Dockerfile Instructions](#main-dockerfile-instructions)
+  - [Let's Give it a Go](#lets-give-it-a-go)
+- [Docker Image](#docker-image)
+  - [Remember when creating a docker image](#remember-when-creating-a-docker-image)
+- [Main docker commands](#main-docker-commands)
+- [ECS](#ecs)
+  - [ECS Cluster](#ecs-cluster)
+  - [ECS Service](#ecs-service)
+  - [ECS Task Definition](#ecs-task-definition)
+- [ECR](#ecr)
+- [Main Take Aways](#main-take-aways)
+
+## Containers
+
+### What is a container?
+A container is a standard unit of software that packages up code and all its dependencies so the application runs quickly and reliably from one computing environment to another. A container is supposed to run a single main process. Once the process is completed/stoppped, the container will exit.
+<p align="center">
+    <img src="assets/container01.jpg" width="900">
+</p>
+
+### Difference between Virtual Machine and Container
+Both the virtual machine and the container have resource isolation, but they work in a different way. While the virtual machine virtualizes the hardware, the container virtualizes the operating system. This makes the container a lot more portable and efficient.
```

Comment:
The content is good. The presentation is too verbose, try to break lines to split sentences. Easier to read and to follow.

It applies for all sections.

Reply:
fixed.